modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceImplicitGrantTestCase.java

```diff
@@ -58,16 +60,30 @@
 	private String consumerSecret;
 
 	private CloseableHttpClient client;
+	private final String username;
+	private final String userPassword;
+
+	@DataProvider(name = "configProvider")
+	public static Object[][] configProvider() {
+		return new Object[][]{{TestUserMode.SUPER_TENANT_ADMIN}, {TestUserMode.TENANT_ADMIN}};
+	}
+
+	@Factory(dataProvider = "configProvider")
+	public OAuth2ServiceImplicitGrantTestCase(TestUserMode userMode) throws Exception {
+
+		super.init(userMode);
+		AutomationContext context = new AutomationContext("IDENTITY", userMode);
+		this.username = context.getContextTenant().getTenantAdmin().getUserName();
+		this.userPassword = context.getContextTenant().getTenantAdmin().getPassword();
+	}
 
 	@BeforeClass(alwaysRun = true)
 	public void testInit() throws Exception {
-		super.init(TestUserMode.SUPER_TENANT_USER);
+
 		logManger = new AuthenticatorClient(backendURL);
 		adminUsername = userInfo.getUserName();
 		adminPassword = userInfo.getPassword();
-		logManger.login(isServer.getSuperTenant().getTenantAdmin().getUserName(),
-				isServer.getSuperTenant().getTenantAdmin().getPassword(),
-				isServer.getInstance().getHosts().get("default"));
+		logManger.login(username, userPassword,	isServer.getInstance().getHosts().get("default"));
```

Comment:
change this line to 
        sessionCookie = logManger.login(username, userPassword, isServer.getInstance().getHosts().get("default"));